src/structures/idt.rs

```diff
@@ -918,6 +931,57 @@ bitflags! {
     }
 }
 
+// https://wiki.osdev.org/Exceptions#Selector_Error_Code
+/// Describes a segment selector error code.
+#[repr(transparent)]
+pub struct SelectorErrorCode {
+    flags: u64,
+}
+
+impl SelectorErrorCode {
```

Comment:
Could we add a `const fn` new method that took a u64 and returned either a value or an option?

Reply:
Since only 16 of the bits do anything. Would it be better to take in a u16 that produce an always valid `SelectorErrorCode`? Or take a u64 that might have bits set in the reserved section and produce an Option? Otherwise we could take a u64 and wipe the most significant 48 bits.

Reply:
I think we should try to have `SelectorErrorCode::new()` work a lot like [`PageFaultErrorCode::from_bits()`](https://docs.rs/x86_64/0.14.3/x86_64/structures/idt/struct.PageFaultErrorCode.html#method.from_bits). It should take in a `u64` and return a `Option<SelectorErrorCode>`, where `None` is returned if any reserved bits are set.